---
title: "Lab Meeting"
author: |
  | Karsten Bach
date: '`r Sys.Date()`'
output:
  beamer_presentation:
    font: "structurebold"
    fig_caption: FALSE
    includes:
     in_header: header.tex
jont-family: 'Helvetica'
---

# Understanding Breast Cancer Heterogeneity
![](img/CellOfOrigin.pdf)

# How to identify the cell of origin
\columnsbegin
\column{0.6\textwidth}
\begin{enumerate}
\item Expression of marker genes
\item Promoter driven oncogenes
\item Lineage tracing
\item Single-cell Genomics
\end{enumerate}
\column{0.5\textwidth}
![](img/CellOfOrigin2.pdf)
\columnsend

# Motivation
\begin{center}
\includegraphics{img/CellOfOrigin3.pdf}
\end{center}

# Experimental Setup
![](img/setup0.pdf)

# Experimental Setup
![](img/setup1.pdf)

# Experimental Setup
![](img/setup2.pdf)

# Experimental Setup
![](img/setup3.pdf)

# Experimental part
## Conclusions
- 8 samples is the maximum with this protocol 
- Cell number is the main limitation
    - 20 - 100k per sample after ~20min sort
- Accurate cell counting is important

## How to improve
- Optimize digestion 
- Sort in small volume to avoid spinning \& straining
- Count every sample

# Computational Analysis
\vspace{15pt}
\hspace{50pt}
![](img/CompAnalysis.pdf)
```{r echo=FALSE, message=FALSE, cache=TRUE, warning=FALSE, fig.keep="none"} 
library(knitr)
setwd("../src/")
source(purl("QCAnalysis.Rmd",output=tempfile(),quiet=TRUE),
       echo=FALSE, verbose=FALSE, prompt.echo=FALSE)
setwd("../presentations/")
```

# Overall cell numbers
```{r echo=FALSE, message=FALSE}
library(knitr)
library(dplyr)
library(ggplot2)
library(viridis)
kable(t(as.matrix(table(pD$SampleID))), caption="Number of sequenced cells per sample", format="pandoc")
kable(t(as.matrix(table(pD$Condition))), caption="Number of sequenced cells",format="pandoc")
```

# Libraries from lactation sample have low complexity
- [Lactation Sample 2](../data/CellRangerReports/SampleF1_summary.html)
- [Virgin Sample 2](../data/CellRangerReports/SampleG1_summary.html)

# QC stats - Number of Genes detected
```{r echo=FALSE}
genesDetected
```

# QC stats - Number of Genes Threshold
```{r echo=FALSE, warning=FALSE, message=FALSE}
gdHist
```

# QC stats - Library size
```{r echo=FALSE}
LibrarySize
```

# QC stats - Library size Threshold
```{r echo=FALSE, warning=FALSE, message=FALSE}
libSizeHist
```

# QC stats - Most highly expressed genes
```{r echo=FALSE}
topGenes
```

# QC stats - Ratio of Mitochondrial Genes
```{r echo=FALSE}
cellViability
```

# What are we left with
```{r echo=FALSE, warning=FALSE}
kable(table(pD$Condition,pD$PassAll), caption="Cells retained after QC", format="pandoc")
```

# Presumably fair representation of lactation sample
```{r echo=FALSE, warning=FALSE}
tsnPlot
```

# The "data"
```{r echo=FALSE}
knitr::knit_hooks$set(mysize = function(before, options, envir) {
  if (before) 
    return(options$size)
})
```
```{r echo=TRUE, mysize=TRUE, size="\\footnotesize", comment=""}
m.filtered[1:5,1:2]
dim(m.filtered)
```

# Highly variable Genes
```{r echo=FALSE, warning=FALSE}
genes2lbl <- c("Acta2", "Elf5", "Csn2", "Wap","Krt14", "Pgr", "Prlr", "Gata3")
lblDat <- filter(fD.filtered, symbol %in% genes2lbl)
V2 <- ggplot(arrange(fD.filtered,highVar), aes(x=meanExpression,y=CV2, color=highVar)) +
    geom_point(data=lblDat, aes(x=meanExpression, y=CV2), pch=1, size=4) +
    geom_point(alpha=0.8) +
    geom_label(data=lblDat, aes(x=meanExpression, y=CV2, label=symbol), nudge_y=0.1) +
    scale_x_log10() +
    scale_y_log10() +
    theme_bw()
V2
```

# General structure
```{r echo=FALSE, warning=FALSE}
library(viridis)
pcaPlot <- ggplot(pD.filtered, aes(x=PC1, y=PC3, color=Condition, shape=Replicate)) +
    geom_point(alpha=0.8,size=2) +
    theme_bw() 
pcaPlot
```

# General structure - Milk Proteins
```{r echo=FALSE, warning=FALSE}
pcaPlot <- ggplot(pD.filtered, aes(x=PC1, y=PC3, color=log2(Csn2+1), shape=Replicate)) +
    geom_point(alpha=0.8,size=2) +
    scale_color_viridis() +
    theme_bw() 
pcaPlot
```

# General structure - Basal Marker
```{r echo=FALSE, warning=FALSE}
pcaPlot <- ggplot(pD.filtered, aes(x=PC1, y=PC2, color=log2(Acta2+1), shape=Replicate)) +
    geom_point(alpha=0.8,size=2) +
    scale_color_viridis() +
    theme_bw() 
pcaPlot
```

# General structure - tSNE
```{r echo=FALSE, warning=FALSE}
Krt14 <- filter(fD.filtered, symbol=="Krt14") %>% .$id
pD.filtered$Krt14 <- t(m.norm)[,Krt14]
tsnePlot <- ggplot(pD.filtered, aes(x=tSNE1, y=tSNE2, color=Condition, shape=Replicate)) +
    geom_point(alpha=0.8,size=2) +
    theme_bw()
tsnePlot
```

# General structure - Basal cells
```{r echo=FALSE, warning=FALSE}
tsnePlot <- ggplot(pD.filtered, aes(x=tSNE1, y=tSNE2, color=log2(Krt14+1), shape=Replicate)) +
    geom_point(alpha=0.8,size=2) +
    scale_color_viridis() +
    theme_bw()
tsnePlot
```

# General structure - Cd74 cells?
```{r echo=FALSE, warning=FALSE}
Cd74 <- filter(fD.filtered, symbol=="Cd74") %>% .$id
pD.filtered$Cd74 <- t(m.norm)[,Cd74]
tsnePlot <- ggplot(pD.filtered, aes(x=tSNE1, y=tSNE2, color=log2(Cd74+1), shape=Replicate)) +
    geom_point(alpha=0.8,size=2) +
    scale_color_viridis() +
    theme_bw()
tsnePlot
```

# Clustering
![](img/Clustering.pdf)

# Outlook
## Analysis
- Identify clusters and relate them to cell types
- Differentially expressed genes within each cell type
- Differentiation trajectories
    - e.g. in the luminal department
- Testing specific hypotheses 

## Future
- Plan experiment for tumor development

#

```{r echo=FALSE, warning=FALSE}
Bcl11a <- filter(fD, symbol=="Bcl11a") %>% .$id
pD.filtered$Bcl11a <- t(m[,as.character(pD.filtered$barcode)])[,Bcl11a]/pD.filtered$sf
tsnePlot <- ggplot(pD.filtered, aes(x=tSNE1, y=tSNE2, color=log2(Bcl11a+1), shape=Replicate)) +
    geom_point(alpha=0.8,size=2) +
    scale_color_viridis() +
    theme_bw()
tsnePlot
```
