\documentclass{article}
\usepackage[backend=bibtex,
	    natbib=true,
	    autocite=superscript,
	    style=nature,
	    url=true,
	    sorting=none,
	    isbn=false]{biblatex} 
\addbibresource{../../paper/references.bib}
\usepackage{hyperref}
\usepackage{graphicx, subcaption}
\usepackage{url}
\usepackage[a4paper]{geometry}
\usepackage{pdfpages}
\renewcommand{\thesubfigure}{\Alph{subfigure}}
\newcommand{\comment}[1]{\textit{\textcolor{blue}{Comment: #1}}}
\begin{document}
\includepdf{letter.pdf}

%---------------------Index-Swapping--------------------------------------
\section*{Index swapping}
After the submission of our manuscript \citeauthor{Sinha2017} reported a spreading-of-signal from multiplexed samples on the HiSeq 4000 \autocite{Sinha2017}.
This is referred to as index-swapping and leads to wrongly assigning reads to samples during de-multiplexing.
The full scope of index-swapping has been extensively documented in \autocite{Sinha2017} to which Illumina has responded in the form of a white paper \autocite{Illumina}.
At the time of submission we were unaware of the mere possibility of this artifact and have thus only now analysed our data in this regard.
We found that index-swapping although substantially affecting our data has not altered our biological conclusions.
To show this we have re-sequenced the library on a HiSeq 2500, which is almost not affected by index-swapping.
In addition, we have performed an independent repeat experiment, meaning that we have prepared fresh libraries from new samples and sequenced again on a HiSeq 2500.
In both instances our biological conclusions held, which we see as strong support for the validity of our findings.

%-------------------------------------------------------------------------
\subsection*{Introduction}
As mentioned above the underlying mechanisms and effects on de-multiplexing are extensively covered in \autocite{Sinha2017}, hence we only briefly introduce the topic to improve readability.
Index-swapping is defined as the priming and extension of library molecules by free indexing primers.
Mispriming of free indexing primers and extensions lead to the duplication of reads and incorporation of a new indexing sequence (\autoref{F1}A).
Essentially, this allows information from one sample to bleed into other samples as exemplified in plate based scRNA-seq data from \autocite{Sinha2017}, where reads bleed into the rows or columns due to the swapping of one of the two indices (\autoref{F1}B).\\
The library design of 10x libraries differs in the regard that the Illumina indices are used to identify samples (in our case e.g. ``Virgin Replicate 1'') instead of cells (\autoref{F1}C).
Cells are distinguished by a 14bp cellular barcode which is drawn at random from a pool of $\sim$750,000 possible barcodes \autocite{Zheng2017}.
In the event of index-swapping we would see the same read appearing in two or more samples, in which case the sequence of the cell barcode would also be identical.
Hence, all reads that have swapped between samples can only be present in cells that share barcode sequences between samples.
Conceptually, we can distinguish three potential scenarios.\\
In the first scenario, two samples contain two cells that share the same barcode by pure chance, which would mean that reads between the two cells could be swapped between samples in either direction.
The impact of this is however expected to be small due to the small chance of observing one out of $\sim$750,000 possible barcodes in two samples.
In the second scenario, only one of two samples (sample A) contains a cell with a specific barcode whereas the other does not (sample B).
In this case, the swapping is mostly uni-directional, where sample A bleeds into B.
Sample B only contains some background levels of reads for this barcode and the absolute number of reads swapped into the other sample should be insignificant.
The third possibility arises when reads from sample A bleed enough into sample B to provide sufficient reads to create a miniature copy of itself in sample B.
The amount of bleedthrough necessary to create copies depends on the library size distribution of the cells present in the ``recipient'' sample due to how CellRanger distinguishes between cells and background.\\
In summary, the first scenario is a rare event that only affects very few cells.
The effect of the second case on the data is negligible as this might only increase the general noise in the data to a small extent.
The third scenario can be more severe depending on the library size distribution of the samples that were run together on the HiSeq 4000.

\begin{figure}
    \begin{subfigure}[c]{0.5\textwidth}
	\includegraphics[width=\textwidth]{img/swapping.png}
    \caption{}
    \end{subfigure}
    \begin{subfigure}[c]{0.5\textwidth}
	\includegraphics[width=\textwidth]{img/swapping2.png}
    \caption{}
    \end{subfigure}
    \begin{subfigure}[c]{0.5\textwidth}
	\includegraphics[width=\textwidth]{img/library.png}
    \caption{}
    \end{subfigure}
    \caption{Overview of index-swapping.
	(\textbf{A}) illustrates the process of index-swapping during cluster generation with the ExAmp chemistry.
	(\textbf{B}) shows the spreading-of-signal observed when a cell expresses high levels of a gene that is not expressed in other cells on the plate.
	Swapping of a single barcode on either end then leads to the cross-hair pattern.
	(\textbf{C}) illustration of the library generated from the 10x, Illumina barcodes that swap are highlighted. 
	Figures \textbf{A} and \textbf{B} are adopted from \autocite{Sinha2017}.
    }
    \label{F1}
\end{figure}

\subsection*{Impact on the presented data}
\paragraph{Index swapping occurred}
First, we noted that the samples in our data share a high number of barcodes (\autoref{F2}A).
In a one-by-one comparison between samples all possible comparisons show a higher number of barcodes shared than expected by chance (\autoref{F2}B).
This suggests the presence of the third scenario, where index-swapping creates ``pseudo-cells''.
If this is the case we would also expect cells that share barcodes across samples to be highly similar, which again would otherwise not be expected.
Indeed, we found that cells sharing the same barcode were almost always part of the same cluster, suggesting a high similarity (\autoref{F2}C).
Lastly, we also identified a pattern per barcode where one cell has substantially more mRNA molecules than the remaining cells for that barcode (\autoref{F2}D). \\
All of this suggested to us that index-swapping occurred and has led to the identification of ``pseudo-cells'' that are in fact only copies of other cells from other samples.\\

\begin{figure}
    \begin{subfigure}[c]{0.5\textwidth}
	\includegraphics[width=\textwidth]{img/p1.pdf}
    \caption{}
    \end{subfigure}
    \begin{subfigure}[c]{0.5\textwidth}
	\includegraphics[width=\textwidth]{img/p2.pdf}
    \caption{}
    \end{subfigure}
    \begin{subfigure}[c]{0.5\textwidth}
	\includegraphics[width=\textwidth]{img/p3.pdf}
    \caption{}
    \end{subfigure}
    \begin{subfigure}[c]{0.5\textwidth}
	\includegraphics[width=\textwidth]{img/p4.pdf}
    \caption{}
    \end{subfigure}
    \caption{Index swapping compromised parts of the presented data.
	(\textbf{A}) Number of barcodes detected in each samples, colored by the number of times each barcode appears in the entire dataset.
	(\textbf{B}) Volcano-plot based of the number of shared barcodes versus the P value to observe at that many or more barcodes shared between any two samples.
	P values are based on a hypergeometric test for all possible two-sample comparisons. \comment{add more detail about test?}
	(\textbf{C}) Histogram of the fraction of cells with the same barcode that were identified in our analysis as being part of one cluster.
	(\textbf{D}) Kernel density estimates of the distribution of ratios between the largest cell and the mean of the remaining cells for each barcode that was shared between multiple samples.
    }
    \label{F2}
\end{figure}

\paragraph{Cells identified in the lactation sample derive from index-swapping}
\autoref{F1}A suggests that the lactation samples in particular were severely affected with only a single cell having a unique barcode.
In addition, 313 out of all 314 non-unique barcodes across the entire dataset appeared at least once in the lactation sample and in 298 case was the cell with the largest library size not coming from the lactation sample (\autoref{F1}A, \autoref{F3}A).
This suggests that these were the samples that were mainly involved in index swapping by soaking up reads from other samples.
Further, we can look at the pattern of index-swapping in more detail, assuming that for each barcode the cell with the largest number of UMIs is the cell that gave rise to ``pseudo-cells'' with fewer UMIs.
Here it becomes evident again that in the majority of cases one or both lactation samples received reads from swapping (\autoref{F3}B).
Together, this suggested to us that the vast majority of cells in the lactation samples are ``pseudo-cells'' derived from index-swapping.
The most likely explanation for this is that the cells form the lactation sample either did not lyse properly or lysed before they were encapsulated in the beads.
This explains the low amount of cDNA we observed for these samples (data not shown).
On a HiSeq 2500 this would have led to the detection of only few cells.
However, due to the index-swapping on the HiSeq 4000 these samples acted as a pool that soaked up index-swapped reads from other samples.\\
\autoref{F3}B also shows that in the few cases were reads were not swapped into the lactation samples, the lactation samples themselves bled into the other samples.
In total we found 17 barcodes that appeared to stem from true cells in the lactation samples, notably all cells of cluster 7 and 8 were part of this (\autoref{F3}C).
The fact that the cells from lactation were highly likely to bleed into other samples can be explained by the high relative concentration of their cDNA.
As all cDNA were pooled at equal amounts for sequencing the lactation samples contained much more cDNA per cell than the other samples, increasing the likelihood of index-swapping.
This also explains why the ratio of total number of unique molecules in \autoref{F1}C is low when the reads came from lactation.\\

In summary, the data we presented was compromised by index swapping, a technical artifact specific to the Illumina HiSeq 4000 that leads to a spreading-of-signal.
This had a substantial effect on the lactation samples, which - considering the new evidence - are most likely to have been almost completely avoid of any true cells.
This means that all inference based on these two sample are meaningless. 
In addition, two of the identified clusters (C7, C8) are now also questionable as they are made up of much fewer cells than previously thought.\\
In order to gauge how much this has impacted the biological conclusion we drew from our analysis we re-sequenced the same cDNA library on a HiSeq 2500.
Furthermore, we have repeated the entire experiment taking some of the reviewers' comments into consideration in order to be fully confident of our statements.
In the following we will summarize the re-analysis of the HiSeq 2500 data from the first experiment.
The second experiment can then be found in the revised version of the manuscript.




\begin{figure}
    \begin{subfigure}[c]{0.5\textwidth}
	\includegraphics[width=\textwidth]{img/p5.pdf}
    \caption{}
    \end{subfigure}
    \begin{subfigure}[c]{0.5\textwidth}
	\includegraphics[width=\textwidth]{img/p6.pdf}
    \caption{}
    \end{subfigure}
    \begin{subfigure}[c]{0.5\textwidth}
	\includegraphics[width=\textwidth]{img/p7.pdf}
    \caption{}
    \end{subfigure}
    \caption{Lactation sample contained no cells.
    }
    \label{F3}
\end{figure}

\subsection*{Biological conclusions remain unaffected}
\subsection*{Conclusions}
\newpage
%---------------------Addressing comments--------------------------------------
\section*{Reviewers' Comments}
\subsection*{Reviewer \#1}
\textbf{1. Can it be excluded that Cluster 9 is a non-epithelial contamination? For instance, microvascular or perivascular cells? Cluster 9 cells, similar to myoepithelial cells, express some smooth muscle markers at low/medium level, however, they hardly express any epithelial marker.}\\
\ldots

\textbf{2. Samples from nulliparous and post-involution mice - NP1, NP2, PI1 and PI2. It is not mentioned at what stage of estrus cycle these mice were sacrificed, yet, it is known that gene expression is differentially regulated at different estrus stages. The data shown in Supplementary Figure 2a reveal significant differences in cell distribution between NP1 and NP2, as well as between PI1 and PI2. In pregnancy and lactation, hormonal levels should not differ significantly between individuals 1 and 2. Consistently, the samples 1 and 2 from pregnant and lactating mice, look similar within each stage - L1 and L2 samples, as well as G1 and G2 samples perfectly overlap in Supplementary Figure 2a. Ideally, cycling (NP and PI) mice should have been synchronized prior to cell sorting. This issue should be mentioned in the discussion.}\\
\ldots

\textbf{3. Why is Acta2 relatively high in some luminal clusters shown in Supplementary Figure 2c, – for instance, in Cluster 2, secretory luminal population? Is it “experimental noise”? Could the authors comment.
Page 7, lines 148-149.”… as well as transcription factors that have not previously been associated with luminal differentiation such as Creb5, Hey1…”. This statement is inexact. On the contrary, Hey1 is a well established marker of luminal progenitors (Bouras et al., 2008. Notch Signaling Regulates Mammary Stem Cell Function and Luminal Cell-Fate Commitment, Cell Stem Cell, 3: 429).
Page 9, lines 203-205. In contrast to authors’ statement, it is not clear how the results of this work “might help to explain some of the conflicting results from lineage tracing studies”. The contradictions in lineage tracing data concern mostly the bipotency of basal stem cells, whilst, as the authors mention, the results of this work rather support the concept of lineage-restricted progenitor/stem cells in both compartments (Figure 2a).}\\
\ldots

\textbf{4. Page 4, lines 87-88 “…from two independent mice.” What does “independent” stand for? Littermates?}\\
\ldots

\textbf{5. Page 9, lines 189-190 “…luminal progenitor cells maintain memory of having undergone gestation and involution.” Maybe, “gestation and lactation”?}\\
\ldots

\subsection*{Reviewer \#2}
\textbf{1. A major consideration with this paper is the use of Epcam to sort epithelial cells. Prior seminal publications (eg doi:10.1038/nm.2000) using human cells demonstrate an important contribution of Epcam negative epithelial cells to stem / progenitor activity in the mammary gland. What is the evidence that Epcam quantitatively recovers mouse mammary epithelia? The consequence of this methodology is the possible exclusion of subsets of mammary epithelial cells from the data capture.}\\
\ldots

\textbf{2. Very few studies have used this system to sample diverse epithelial cell types, so it is not yet clear what sampling bias might be present in this method. The authors should compare the proportion of cell types represented in the single cell data with equivalents identified by flow cytometry at these developmental stages to address sampling bias.}\\
\ldots

\textbf{3. The authors conducted analysis on biological replicates, however this is not described in the results section. This is an important consideration with a new technology and the variability between replicates should be appropriately quantitated and addressed in the results. }\\
\ldots

\textbf{4. Given that the authors see this dataset as a resource, data should be made available for download.}\\
\ldots

\textbf{5. Terminal End Buds (TEBs) are a critical mediator of pubertal development, with unique gene expression and function. TEBs start to disappear by the 8 weeks of age point used for nulliparous animals, so these cell types may not have been sampled. Do the authors observe TEBs at this time point?}\\
\ldots

\textbf{Minor issues: \\
1. The link to the scripts on github is not operational \\ 
2. Was estrous stage controlled, measured or considered for the nulliparous mice? This will impact gene expression. \\
3. Please include a summary table of the number of cells captured from each animal, both total and in each cluster \\
4. The description of Fig 1 could use more cross referencing to known markers of each subset. \\
5. The colours of dots in each figure are very hard to discriminate, especially for those with diminished colour vision. Please try to choose a higher contrast palette or the use of patterns to distinguish groups. \\
6. Please label all axes, Eg. Fig A,D,E \\
7. Please conduct pathway and ontology analysis of the genes changing through pseudotime- described in lines 144-151. \\
8. C7 doesn’t appear in Fig 2A. Please correct. \\
9. The genes referred to in lines 134-136 can’t all be seen in Fig 1d.}\\
\ldots


\subsection*{Reviewer \#3}
\textbf{1. The authors need to do independent experiments to confirm that the lactation stage cells really express so low number of genes. I am not convinced that an individual cell at L1 stage only express 400 genes. The authors should did single cell RNA-seq using SMART-seq2 protocol for all of these four stages of cells, probably five to ten single cells for each stage and check if lactation stage cells really express so low number of genes compared with the other three stages of cells.}\\
\ldots

\textbf{2. For Cluster 9, it seems that only 15 cells are there. And for Cluster 8, it seems that only 26 cells are there. Are they real clusters or just doublets? The authors should analyze further of these two clusters. For example, are C8 cells are really intermediate state of C2 \& C5? Or they are merely doublets of C2+C5 cells? Probably the authors should capture all of the differentially expressed genes between C2 and C5 and check if C8 cells in general express these genes at 1/2 of the level of sum of C2+C5. Or maybe do some immunostaining to see if the cells double positive for C2 and C5 markers are really there.}\\
\ldots

\textbf{3. The authors should give the details of the filtration of the poor quality cells: exactly how many genes detected, how many UMI detected, what percentage of reads mapped to mitochondrial genes. Are they the same for all of the four stages of cells? Or different criteria for different stages of cells?}\\
\ldots

\printbibliography
\end{document}
